### PR TITLE
Serverless contact form

### PR DESCRIPTION
**Pull request by tpartington**
use api-gateway and lambda for the cevo-website contact form.
the api-key is used to enforce rate limiting on the server side so it's not a secret.

this implementation will throttle requests to 1 per second. and will only allow 50 requests a day.

still to come is monitoring and alerting so we know when this is broken.
we should also disable the submit button once it has been clicked to avoid duplicate submissions while the async request is in flight.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -21,6 +22,11 @@ default: build
 clean:
 	rm -fr ./public
 
+serverless:
```
Comment (stevemac007) [inline Makefile]: Am I reading this right that the deploy needs to be manually invoked?  Or is it wired into being deployed via TravisCI?

File: config.yaml [hunk 2]
```diff
@@ -143,7 +143,8 @@ params:
         warning: Please enter your phone number.
     subtitle: We will get back to you within 24 hours.
     title: Contact us
-    postURL: https://formspree.io/xrqqlwqm
+    postURL: https://7opj4a2h2k.execute-api.ap-southeast-2.amazonaws.com/production
```
Comment (stevemac007) [inline config.yaml]: Is there a way (and even should we) put a cevo.com.au domain name on this?  Means we can redirect it from the magic API GW unique name if we need to change it.

File: config.yaml [hunk 3]
```diff
@@ -143,7 +143,8 @@ params:
...
+    postApiKey: Sp2CN9bhAa27PByMgGYsA2rJwVAu4ZCB8xeagzrt
```
Comment (stevemac007) [inline config.yaml]: Where does this magic key come from?

Can we add some commentary to the README explaining this stuffz?

File: serverless-contact-form.js [hunk 4]
```diff
@@ -0,0 +1,82 @@
+// Code modified from the example given at
+// https://codehabitude.com/2016/04/05/forms-to-emails-using-aws-lambda-api-gateway/
+
+var AWS = require('aws-sdk')
+const querystring = require('querystring');
+
+// Set the region 
+AWS.config.update({region: 'us-west-2'});
```
Comment (stevemac007) [inline serverless-contact-form.js]: Cheapest region?

File: serverless-contact-form.js [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+                }
+            },
+            Subject: {
+                Data: 'Website Contact From: ' + contact_name,
+                Charset: 'UTF-8'
+            }
+        },
+        Source: SENDER
+    }
+    ses.sendEmail(params, done)
```
Comment (stevemac007) [inline serverless-contact-form.js]: Noob question here - How does the SES permission stuffz work?  Is it part of the IAM role attached to the Lambda?

Is there an existing SES endpoint for prod?  Do we need to deploy this to `beta` too - but have it go to a different mailbox?

File: config.yaml (hunk 3 above)
Comment (stevemac007) [reply config.yaml]: DOH! read up!!

File: Makefile (hunk 1 above)
Comment (tpartington) [reply Makefile]: This implementation needs to be run manually.
I'm looking into getting Travis to handle the deployment of the stack as well.

File: config.yaml (hunk 2 above)
Comment (tpartington) [reply config.yaml]: yep, we can route53 it. Is it worth the effort though?

File: serverless-contact-form.js (hunk 4 above)
Comment (tpartington) [reply serverless-contact-form.js]: closest region to Aus that supports SES

File: serverless-contact-form.js (hunk 5 above)
Comment (tpartington) [reply serverless-contact-form.js]: The role attached to the Lambda function needs to have the `ses:SendEmail` permission, and in SES you need to validate your address (they send you an e-mail, you click on a link).

This implementation will use the same serverless stack (in cevo-prod) for both beta and prod.

File: serverless-contact-form.js (hunk 4 above)
Comment (stevemac007) [reply serverless-contact-form.js]: I forget not everything is here..

File: config.yaml (hunk 2 above)
Comment (stevemac007) [reply config.yaml]: Just removes any magic url's - not critical but I see some of our work a good excuse to flesh out patterns we should champion on client site.

File: Makefile (hunk 1 above)
Comment (stevemac007) [reply Makefile]: Missed that in the docco - sorry.

File: serverless-contact-form.js (hunk 5 above)
Comment (stevemac007) [reply serverless-contact-form.js]: As long as that is known / expected that both beta and prod are live emails - thats cool for now.

**Pull request merged**